network/engine/engine_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2021. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package engine
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/nuts-foundation/nuts-node/network"
+	"github.com/nuts-foundation/nuts-node/network/dag"
+	"github.com/nuts-foundation/nuts-node/network/test"
+	"github.com/nuts-foundation/nuts-node/test/io"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestNewEngine(t *testing.T) {
+	testDirectory := io.TestDirectory(t)
+	assert.NotNil(t, NewNetworkEngine(network.NewTestNetworkInstance(testDirectory)))
+}
+
+func TestFlagSet(t *testing.T) {
+	assert.NotNil(t, flagSet())
+}
+
+func TestCmd_List(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mock := network.NewMockNetwork(ctrl)
+	clientCreator = func() network.Network {
+		return mock
+	}
+	mock.EXPECT().ListDocuments().Return([]dag.Document{dag.CreateTestDocument(1)}, nil)
+	cmd := cmd()
+	cmd.SetArgs([]string{"list"})
+	err := cmd.Execute()
+	assert.NoError(t, err)
+}
+
+func TestCmd_Get(t *testing.T) {
```

Comment:
Would this be easier when using the `httptest` server?

Reply:
That would mean using HTTPClient instead of the interface which complicates things; since then you test the CLI command **and** the HTTP client at the same time (as sort of integration test), while the http client has its own unit tests. That's why HTTPClient implementing the engine's Go API (interface) makes things simpler (imo).